src/hooks/useInfiniteScroll/useInfiniteScroll.tsx

```diff
@@ -1,8 +1,18 @@
-import { useCallback, useMemo, useState } from 'react'
+import { useCallback, useEffect, useMemo, useState } from 'react'
 
 const defaultAmount = 20
 
-export const useInfiniteScroll = <T extends any>(array: T[], initialTxsCount?: number) => {
+type UseInfiniteScrollProps<T> = {
+  array: T[]
+  initialTxsCount?: number
+  isScrollable?: boolean
+}
+
+export const useInfiniteScroll = <T extends any>({
```

Comment:
Confirmed we're dealing with too different of structures across usages and `any` is actually valid here ✅